test/malloc_compliance_tests.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Basic tests
+// ISO/IEC 9899:201x 7.22.3 compliance
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "malloc_compliance_tests.hpp"
+#include "test_helpers.h"
+#include "umf/memory_pool.h"
+
+#include "base.hpp"
+using umf_test::test;
+
+//------------------------------------------------------------------------
+// Configurable defs
+//------------------------------------------------------------------------
+
+#define MAX_ALLOC_SIZE (1024 * 1024) // 1 MB
+#define ITERATIONS 100
+#define SRAND_INIT_VALUE 0
+
+//------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------
+
+static inline size_t rand_alloc_size(int max) {
+    srand(SRAND_INIT_VALUE);
+    return rand() % max;
+}
+
+static inline void free_memory(umf_memory_pool_handle_t hPool,
+                               void *ptr[ITERATIONS]) {
+    for (int i = 0; i < ITERATIONS; i++) {
+        umfPoolFree(hPool, ptr[i]);
+    }
+}
+
+//------------------------------------------------------------------------
+// Tests
+// -----------------------------------------------------------------------
+
+// ISO/IEC 9899:TC3 7.20.3.3
+void malloc_compliance_test(umf_memory_pool_handle_t hPool) {
+    void *alloc_ptr[ITERATIONS];
+    size_t alloc_ptr_size[ITERATIONS];
+
+    // Each allocation shall yield a pointer to an object disjoint from
+    // any other object.
+    for (int i = 0; i < ITERATIONS; i++) {
+        alloc_ptr_size[i] = rand_alloc_size(MAX_ALLOC_SIZE);
+        alloc_ptr[i] = umfPoolMalloc(hPool, alloc_ptr_size[i]);
+
+        // Supported under USE_WEAK_ALIGNMENT_RULES macro
+        // For compatibility, on 64-bit systems malloc should align to 16 bytes
```

Comment:
It fails on Windows !?!
https://github.com/oneapi-src/unified-memory-framework/actions/runs/7220175319/job/19672583680?pr=64
https://github.com/oneapi-src/unified-memory-framework/actions/runs/7220509785/job/19673783556